src/Platforms/AbstractPlatform.php

```diff
@@ -3882,6 +3908,132 @@ public function getTimeTypeDeclarationSQL(array $column)
+     * like CREATE TABLE.
+     *
+     * @param mixed[] $column
+     *
+     * @throws Exception If not supported on this platform.
+     */
+    protected function getDateTimeTypeDeclarationSQLSnippet(array $column): string
+    {
+        throw Exception::notSupported(__METHOD__);
+    }
+
+    /**
+     * Obtains DBMS specific SQL declaration snippet to be used to create datetimetz columns in statements
+     * like CREATE TABLE.
+     *
+     * @param mixed[] $column
+     */
+    protected function getDateTimeTzTypeDeclarationSQLSnippet(array $column): string
+    {
+        return $this->getDateTimeTypeDeclarationSQLSnippet($column);
+    }
+
+    /**
+     * Obtains DBMS specific SQL declaration snippet to be used to create time columns in statements
+     * like CREATE TABLE.
+     *
+     * @param mixed[] $column
+     *
+     * @throws Exception If not supported on this platform.
+     */
+    protected function getTimeTypeDeclarationSQLSnippet(array $column): string
+    {
+        throw Exception::notSupported(__METHOD__);
+    }
+
+    /**
+     * Obtains DBMS specific suffix to add to SQL declaration to create datetime columns in statements
+     * like CREATE TABLE, for example ' WITHOUT TIME ZONE'.
+     *
+     * Requires a leading space unless empty.
+     *
+     * @param mixed[] $column
+     */
+    protected function getDateTimeTypeSQLSuffix(array $column): string
+    {
+        return '';
+    }
+
+    /**
+     * Obtains DBMS specific suffix to add to SQL declaration to create datetimetz columns in statements
+     * like CREATE TABLE, for example ' WITH TIME ZONE'.
+     *
+     * Requires a leading space unless empty.
+     *
+     * @param mixed[] $column
+     */
+    protected function getDateTimeTzTypeSQLSuffix(array $column): string
+    {
+        return '';
+    }
+
+    /**
+     * Obtains DBMS specific suffix to add to SQL declaration to create time columns in statements
+     * like CREATE TABLE, for example ' WITHOUT TIME ZONE'.
+     *
+     * Requires a leading space unless empty.
+     *
+     * @param mixed[] $column
+     */
+    protected function getTimeTypeSQLSuffix(array $column): string
+    {
+        return '';
+    }
+
+    /**
+     * Obtains DBMS specific precision specifier to add to SQL declaration to create datetime columns in statements
+     * like CREATE TABLE.
+     *
+     * @param mixed[] $column
+     */
+    protected function getDateTimeTypePrecisionSQLSnippet(array $column): string
+    {
+        $scale = empty($column['scale']) || (int) $column['scale'] === Column::DATETIME_SCALE_NOT_SET
+                 ? '0'
+                 : $column['scale'];
+
+        return '(' . $scale . ')';
+    }
+
+    /**
+     * Obtains DBMS specific precision specifier to add to SQL declaration to create datetimetz columns in statements
+     * like CREATE TABLE.
+     *
+     * @param mixed[] $column
```

Comment:
```suggestion
     * @param array<string, mixed> $column
```

Reply:
I think `mixed[]` is used because of legacy reasons when there was no `array<...>` syntax. But I'm not a maintainer so don't take my word 100%.